### PR TITLE
[mqtt.homeassistant] Trigger HA devices to update discovery information

### PR DESCRIPTION
**Pull request by ccutrer**

### CODE REVIEW

**Comment by ccutrer** [comment]
Note that the "Home Assistant Discovery" shows up on the Settings page under the System Settings heading instead of Add-on Settings, because the UI will only show services in the Add-on Settings section whose ID matches an installed addon, and whose configuration URI matches said binding name as well (https://github.com/openhab/openhab-webui/blob/main/bundles/org.openhab.ui/web/src/pages/settings/menu/addon-section.vue). This doesn't work, because `mqtt.homeassistant` isn't actually an addon, only `mqtt` is.

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java [hunk 1]
```diff
@@ -63,10 +69,13 @@
  *
  * @author David Graeff - Initial contribution
  */
-@Component(service = DiscoveryService.class, configurationPid = "discovery.mqttha")
+@Component(service = DiscoveryService.class, configurationPid = "org.openhab.binding.mqtt.homeassistant", property = Constants.SERVICE_PID
```
Comment (lsiepel) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java:72]: Just wondering if there is a new naming convention for configration pid or is this just your personal preferrence. (i don;t have an opinion on this, but when reviewing other PR's i see differences.

Comment (ccutrer) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java:72]: I couldn't find that discovery.* showed up in the UI anywhere to actually be configurable. So I based it on other @ConfigurableComponent annotations, such as on HomekitImpl. But maybe I just don't know where to look. I agree it would be much better if there was somewhere to configure this alongside other discovery services.

Comment (lsiepel) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java:72]: I would opt to leave it as is. `discovery.mqttha`

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/resources/OH-INF/config/config.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+		https://openhab.org/schemas/config-description-1.0.0.xsd">
+	<config-description uri="binding:mqtt.homeassistant">
+
+		<parameter name="status" type="boolean" required="false">
+			<label>Publish Online Status</label>
+			<default>true</default>
+			<description><![CDATA[
+         Publish <tt>online</tt> to <tt>homeassistant/status</tt> when discovering Home Assistant
+         things in order to trigger devices to publish up-to-date discovery information.
+				 You may want to disable this if you're also running Home Assistant.
```
Comment (lsiepel) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/resources/OH-INF/config/config.xml:15]: _You may want to disable this if you're also running Home Assistant._

Isn't it a common scenario that Home Assistent co-exists next to openHAB when this binding is in use? What would the implications be, now that this is defaults to true and this disovery is run?

Comment (ccutrer) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/resources/OH-INF/config/config.xml:15]: I'm not sure I'd say it's "common". I think it's common if you're evaluating both, or transitioning from one to the other. I did it for a while when I was evaluating HA for work reasons a long time ago. I recently started it back up in order to validate some contributions to zwave-js-ui against "reference" Home Assistant. Outside of that, there should be plenty of people that use devices compatible with this binding, but only use openHAB. The main point of the binding is to be able to interact with devices that mainly support Home Assistant.

As to the consequences of having this option enabled if you have do run HA in parallel: minor to none. I currently run 4 "classes" of devices connected to openHAB through this binding (and quite a few devices within each class):
 * ESPHome
 * zigbee2mqtt
 * zwave-js-ui
 * Integrations written myself in Ruby using my own library (a couple of which are fairly popular, but the vast majority of users besides myself use Home Assistant, not openHAB)

Of these all except for zwave-js-ui completely ignore the Home Assistant status topic, and simply publish their discovery information as retained. zwave-js-ui can optionally do that, but can also publish it non-retained on demand each time the status topic receives "online" (as triggered by this PR). I'm considering doing the same with my Ruby library, simply to keep the discovery information current for when things are removed. I'm not aware of any device that actually monitors the status topic for "offline" in order to take an action when Home Assistant is offline, but I could conceivably see something like a hypothetical "Remote Home Assistant Integration" (analogous to the Remote openHAB Binding, but to my knowledge doesn't actually exist) or a NodeRED running in conjunction with Home Assistant caring about it. This PR does _not_ publish "online" as retained, so anything initially connecting to MQTT will get the proper status of Home Assistant. The only scenario where it would be wrong is if Home Assistant is not running for some reason, and a discovery is triggered in openHAB (manually, or when openHAB starts). Home Assistant itself does not monitor its own status topic, so openHAB would not be directly interfering with Home Assistant. HA will just receive extra updates to discovery information if openHAB triggers it additionally, which is not harmful.

tl;dr: it's likely a very small audience that would ever even notice any impact from this, and that impact is very unlikely to be negative, thus why I chose to default it to on. That said, I'm amenable to softening the language here. Something like "If you run Home Assistant _and_ other services that depend on knowing if Home Assistant is not running by means of MQTT, then it's possible for those services to be out-of-sync with the actual status of Home Assistant, and you may want to disable this." WDYT?

Comment (lsiepel) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/resources/OH-INF/config/config.xml:15]: First of all thanks for this clear outline of the effects!
With all that in mind, the current description seem a decent indication for affected users to tinker with it or not. Your suggested text is also fine, i leave the final description up to you.

**Review by lsiepel — APPROVED** [review]
Thanks, LGTM!
